src/ruleset.ts

```diff
@@ -775,24 +772,152 @@ export default {
       then: {
         function: pattern,
         functionOptions: {
-          match: "/^https:/",
+          notMatch: "/^http:/",
```

Comment:
What's the reason for this? What if I put something like xyzzy://hello.com 🤔

Reply:
This came from this conversation: https://github.com/stoplightio/spectral-owasp-ruleset/issues/27

Basically the rule is saying "dont use http://" and not saying "you must use this one and only protocol", because that also restricts the use of relative paths, which is not inherently secure or insecure, it just depends. 

Pointing people to `xyzzy://hello.com` is not insecure, at least it's not a problem for the people managing the API. The people managing the API could put something nefarious in to open things up on an iPhone/Desktop app perhaps, but mostly it's just weird, not insecure.